### PR TITLE
Fix bugs for `switch` and add `deleteProfile` command

### LINKED ISSUE #116: Bug: Message change when type only `sw` or `switch`
Expected behavior of switch should be `switch {profile-name}`, currently typing single `switch` or `sw` will simply list all the current addressbook, which is a bit confusing and user may not understand how to use this command. Maybe it would be better if the app can throw a parser exception to require user to input new addressbook name and give the addressbook list in current message for the user to choose if they only type in `switch` or `sw`
image

### PR DESCRIPTION
**Pull request by weiliann**
Fix bugs in profile switching feature and added a delete profile command.

- update message when only `switch` or `sw` is input, to better guide users.
- profiles can be assumed to be created once a switch command is issued, although the file might only be created when the user exists the application. A normal user should not notice any inconsistency from this.
  - developers testing this should exit the program with `exit` or the 'X' button from the window, and not stop the execution from gradle
- delete profile feature is added to allow the profile feature to be more usable
- advanced users could import files into the data/ directory and use the `switch` command to access it.
  - this only work if their data files have the same format as our system

fixes #116 #143 #144 #145

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public static Set<Role> parseRoles(Collection<String> roles) throws ParseExcepti
     public static Profile parseProfileName(String profileName) throws ParseException {
         requireNonNull(profileName);
         String trimmedProfileName = profileName.trim();
-        if (!Profile.isValidProfile(trimmedProfileName)) {
+        if (!Profile.isValidProfileName(trimmedProfileName)) {
             throw new ParseException(Profile.MESSAGE_CONSTRAINTS);
```
Comment (shizy) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:154]: I might prefer `throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, Profile.MESSAGE_CONSTRAINTS));`
basically also use the "Invalid  command format!" message before the constraint message

**Review by shizy — APPROVED** [review]
Overall LGTM! 
I think the `deleteProfile` command is pretty well implemented, I don't really see any bugs.

Something to potentially improve might just be the error messaging. You can consider adding in the `MESSAGE_INVALID_COMMAND_FORMAT` string to the ParseException that is thrown for `sw` or `delp`, just so the user has more context

Also, remember to add in the new command into the `help` window in a future PR, thanks!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (weiliann) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:154]: Thanks. ive updated the error messages and help window

**Review by superb-sushi — APPROVED** [review]
LGTM!

**Pull request merged**